### PR TITLE
Rebasing on TG [DNM]

### LINKED ISSUE #97: Using screw driver on Engine emitter causes it to turn
**Problem Description**:
When you use a screwdriver on the emitter in the engine room the emitter will turn from west facing to east facing. When the panel is closed again via screwdriver the emitter turns back to the normal direction.

**What did you expect to happen**: 
The emitter to stay facing the original direction

**What happened instead**:
The emitter rotated

**Why is this bad/What are the consequences:**
It's only a visual problem, not too huge an issue other than confusion. 

**Steps to reproduce the problem**:
Started round as engineer, used a screwdriver on the emitter as it was not firing due to low power. Emitter flipped around.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.

### LINKED ISSUE #112: Carpet runtimes if it's over-written by another turf
There's a `spawn` that calls `update_icon`, and if the turf's ref changes in that time, we get a bucket of runtimes.

This can arise when loading space ruins, where turfs have a high turnover rate.

### LINKED ISSUE #205: Nav console window has visual issues with outdated Internet Explorer, and on No Frills mode
**Problem Description**:
Nav console, and others, have issues displaying when Internet Explorer is outdated, or when it is on No Frills mode. This issue is not localised to the FTL-13 server, however greatly affects it due to how console-intensive bridge command positions are.

When IE is outdated, the nav console tends to take upward of a minute to open and can crash the game. The FTL map shows squares instead of circles on the star map, and I was having severe difficulty using the helm.

This was on a fresh Windows 7 Ultimate install, so I believe it was IE8 that was installed. The issue was corrected by installing IE 11.

**System environment**
BYOND build 510, Windows 7 Ultimate, build 7601, x64
EVGA GTX 970 SSC / AMD FX-8350.

For further system environment, see attached .NFO.

[sysinfo_dot-NFO_file.zip](https://github.com/FTL13/FTL13/files/532260/sysinfo_dot-NFO_file.zip)

### LINKED ISSUE #243: Borg name does not count
When you start a round as borg and you set your name in lobby, game does not give you the name you've chosen, it's gives you "Default Borg-[number]"

It should take your name from lobby right?

The borg is already not good "class" to play, low power is a real issue, and this default name is just annoying.

That would be nice if I'm not named "Default Borg", I realy like playing borg

See you next round (y)

### LINKED ISSUE #246: Once killed as a blob, 're-enter corpse' re-enters you into blob mode with no way to re-ghost
**Problem Description**:
When you hit 're-enter corpse' as a blob, you are stuck in the blob view with no way to exit. This could be coded to prevent blobs from suiciding because they didn't want to play.

**What did you expect to happen**:
(I accidentally pressed it, it wasn't intentional.)

**What happened instead**:
It brought me back into blob mode, with no way to exit. The connection dropped and I was able to join as spectator.

**Why is this bad/What are the consequences:**
Users who accidentally press it may be stuck and unable to spectate the round or respawn.
ftl13_alpha_testing_nsv_astraeus_2016-11-06_17-45-05

**Steps to reproduce the problem**:
I was made blob by @Qustinnus as a 'round's fucked let's make it more so' thing, and died. I went to go orbit the SM to check on it, missed, and clicked 're-enter corpse' by mistake. I was put back into the blob mode, and couldn't type `ghost` or `suicide` to re-enter ghost mode.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: I was made blob near dorms/EVA.

### LINKED ISSUE #312: mech power
there's a bug with the mechs where if you take out the cell and put in a new one, the mech will still have the same power level of the prior cell

### LINKED ISSUE #325: Exploit from deconning magazines for crazy metal
**Problem Description**:
When you go to deconstruct ammunition magazines **that were on the station at roundstart - DOES NOT APPLY TO MANUFACTURED AMMO** into the autolathe, instead of giving 2400cm3 of metal for say an autorifle mag, it gives 30,000cm3, same applies to sketchkin magazines.

**What did you expect to happen**: 
Pretty sure it's not meant to do that.  Manufactured ammo magazines give what you expect to get.

**What happened instead**:
Shitloads of free metal

**Why is this bad/What are the consequences:**
Shit's gonna be exploited tae fuckery eventually

**Steps to reproduce the problem**:
Don't tell your friends.

### LINKED ISSUE #343: APC in robotics dissappeared when built but not turned on
Robotics got hit by an enemy round and blew up the APC, I placed the frame, wired it, put the board in then screwdrivered it. As soon as the screwdriver finished the APC disappeared completely, it was not in any examine menus, and was not under the wall when I took the wall down. 

The terminal wires still existed but the APC itself was nothing. I couldn't turn the power on and couldn't build a new APC because it said there already was one.

Not sure what the cause was or if it is repeatable.

### LINKED ISSUE #376: Slimes don't get space parallax
**Problem Description**:
What is the problem? Slimes don't space parallax

**What did you expect to happen**: 
Why do you think this is a bug?They should have space parallax

**What happened instead**:
How is what happened different from what you expected?
The space was normal space
**Why is this bad/What are the consequences:**
Why do you think this is an important issue?
Because MUH PARALLAX
**Steps to reproduce the problem**:
The most important section. Review everything you did leading up to causing the issue.
Be a slime.
**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.
I think brains have this problem too

### LINKED ISSUE #393: Food crafting menu freezes up byond
Food crafting needs to be put into seperate categories like with Yogs and TGs crafting, at the moment it freezes up your byond when opening it and crafting a single food piece.

### LINKED ISSUE #415: Repaired plating not displaying properly
So the loaded MAC was opened, causing an explosion. Engineering borg tries to weld the broken plating, which fixed it variable wise, but did not change the appearance as it still appeared broken. Confirmed it was fixed by successfully putting a floor tile on the plating.

No idea how this occurred, no related admin logs.

### LINKED ISSUE #416: Cryocell Beakers
If you have a beaker in a cryo cell, and then click on the cryo cell with another beaker in your hand, the beaker in your hand will be deleted.

### LINKED ISSUE #429: Being super drunk and getting into a disposal makes shit break
**Problem Description**:
>Get super drunk
>hop in disposal chute
>walk sideways to get thrown out of disposal coaster
>body goes flying, camera stays in disposal

**Steps to reproduce the problem**:
Put two disposal trunks ass to ass, one with a chute so it just spits you out the other side, and drink a lot of vodka

### LINKED ISSUE #445: Crew Monitoring Console crashes game when used
Running 511.1359, no other problems with tgui, players experience this as well.

### LINKED ISSUE #452: Ninja's barely recharge
See title. 

Space Ninja's now get barely any charge from electrical sources. I believe this is occurring due to the power update. Obviously, code practicality wise it's not preferred to make a change, and then have to fix things because of it, but if we don't want to change the power revamp, we can just adjust how ninja's charge.

### LINKED ISSUE #453: Deck of cards in a big disappearing from inventory when merged with a card from outside a bag.
**Description of bug:**
Deck of cards in a big disappearing from inventory when merged with a card from outside a bag.

**What did you expect to happen**: 
I expected the deck to merge as per usual.

**What happened instead**:
The deck of cards disappeared, just leaving the single card I was trying to merge it with.

**Why is this bad/What are the consequences:**
While not the biggest issue, due to the limited decks of cards available on the map (I've only located one so far: the bar) it can get in the way of bar RP or simple card games. 

**Steps to reproduce the problem**:
1. Draw a card from a deck, keep it in one of your hands.
2. Put deck of cards into a bag (I was wearing a grey duffle bag on my back at the time)
3. Ensure the card is in your active hand
4. Drag the deck of cards into the card in your hand
5. **Poof!**

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?):** To my knowledge nothing was different about this specific round that would effect the bug, and I do not know what the gamemode was.

### LINKED ISSUE #468: Golem ship has frequent atmospheric problems.
The golem ship found on lava planets has frequent atmospherics problems, which makes receiving visitors a bit of a hassle. Probably be a good idea to port over those fan things from tg that you can find in ruins. Not sure exactly how they work but I know they fix this issue.

### LINKED ISSUE #488: Glowcaps can grow in space
**Problem Description**:
Glowcaps grow into space
**What did you expect to happen**: 
Pretty sure they shouldn't do that

**Steps to reproduce the problem**:
Grow glowcaps, wait for them to grow, boom, in space

### LINKED ISSUE #501: Cannot turn logs into planks from logs from Lava Planet
**Problem Description**:
What is the problem?

Trying to turn logs into planks, but did not work. The logs are from the Xmas area in Lava Planet

**What did you expect to happen**: 
Why do you think this is a bug?

A message pops up indicating it was successful, yet nothing happens

**What happened instead**:
How is what happened different from what you expected?

Objects stay the same

**Why is this bad/What are the consequences:**
Why do you think this is an important issue?

It is not important, but it means I could not make wood items/furniture

**Steps to reproduce the problem**:
The most important section. Review everything you did leading up to causing the issue.

Tried using a hatchet on logs on the ground and in my hands, did not work.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.

Nothing out of the ordinary that I could think. Was in Lava Planet after teleporting there from the ship

### LINKED ISSUE #503: Cloning Pod Print Bug
**Problem Description**: When you click on the cloning pod as a ghost, it decides to print out the 'Examine' Text twice.
What is the problem?

**What did you expect to happen**: For it to print once.
Why do you think this is a bug?

**What happened instead**: It printed it twice.
How is what happened different from what you expected?

**Why is this bad/What are the consequences:** It's not too bad, but it's a bug, bugs should be fixed.
Why do you think this is an important issue?

**Steps to reproduce the problem**: Click on the cloning pod as a ghost
The most important section. Review everything you did leading up to causing the issue.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us. Nothing interesting was going on, I was not doing anything at the time, apart from checking the status of a clone. And I can't tell you anything else.

### LINKED ISSUE #537: Clown ID is always a generic name and not the custom name
watch as the clown eludes the entire sec force and AI under the guise of another clown

### LINKED ISSUE #554: Purge AI module prints, but doesn't appear. Custom Core AI module prints as a standard Core AI Module.
**Problem Description**:
Purge AI module successfully prints, but doesn't appear on top of circuit printer.

Custom core Ai module prints a standard AI module

**What did you expect to happen**: 
Both of these to print, and print correctly.

**What happened instead**:
Because one never appears, the other is incorrect.

**Why is this bad/What are the consequences:**
I couldn't upload britimov.

**Steps to reproduce the problem**:
Attempt to print item in research.

### LINKED ISSUE #559: Door shocks don't work
They just don't, no power in doors to shock. Apparently this is how the powernet works but it'd be lovely if it worked right.

### LINKED ISSUE #651: Runtime on one of the ruins
image

One of those two ruins, probably the DK Excavator 453, causes that runtime. It only happens when that ruin is loaded.

### LINKED ISSUE #666: Job subsystem runtime
image

@ike709

### LINKED ISSUE #674: AI doesn't appear on crew manifest.
The AI isn't on the crew manifest. I only tested this on the PDA crew manifest, not the console crew manifest.

It does appear in the messenger on the PDA, though.

### LINKED ISSUE #699: ALT-TITLES don't show proper icons on secHUDs
**Problem Description**:
SecHUDs don't show the proper icon for alternate titles.
It just shows the grey 'undefined' dot. Speaking of which the Bofficers don't exactly have any either.
@ike709 

Good luck Friendo's

### LINKED ISSUE #701: "Cannot read null.sql_enabled" Runtimes
image

image

### LINKED ISSUE #702: Runtime in preferences.dm, line 125: bad resource file
image

### LINKED ISSUE #703: Runtime in components_base.dm: list index out of bounds
image

### LINKED ISSUE #704: ipintel.dm Runtime(s)
image

I also noticed in a prior round that I could see the ipintel of an admin next to their name when they replied to my admin help, but could not reproduce it during the round I wrote this Issue.

### LINKED ISSUE #707: Runtime in item_attack.dm, line 90: Cannot read null.len
image

### LINKED ISSUE #722: Reagents occasionaly don't react
The primary reason why this is happening is temperature gets set to NaN. This can be fixed by simply setting it to 300. I'll have a temporary fix ready in a bit but the root cause is still unknown.

### LINKED ISSUE #741: Runtime in atmosmachinery.dm, line 132: Cannot execute null.returnpressure()
image

### LINKED ISSUE #742: Runtime in observer.dm, line 366: pick() from empty list
image

### LINKED ISSUE #744: Runtime that creates infinite metal
image

### LINKED ISSUE #749: Bluespace minimoog playing
**Problem Description**:
Drones can play minimoogs at any distance with a bit of help.

**What did you expect to happen**: 
Minimoogs should stop playing when a drone

**What happened instead**:
When the drone is picked up, no check is made to see if the drone is moved, so it keeps playing whatever music is loaded on the minimoog while being in somebody's hand

**Why is this bad/What are the consequences:**
It's not really an important issue, it's just weird that a drone can use it's tiny arms to play music from across the ship. 
**Steps to reproduce the problem**:
1. Be Drone.
2. Get a carbon to pick you up, probably from over 1 tile away from the minimoog
3. Get put on head, enjoy hearing the toy story theme while your new human parades you around the bar

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Nothing major here, it's just kinda derp that it can happen.

### LINKED ISSUE #750: Round start shenanigans
**Problem Description**:
Random teleportation and apparent reequip of starting items as botanist. 

**What did you expect to happen**: 
Well, for starters, there are very few scenarios were you teleport across the ship at round start, and secondly, I'm pretty sure having two PDA's isn't a feature either.

**What happened instead**:
Well...... exactly what i described..... isn't this part of the sheet a little redundant? 

**Why is this bad/What are the consequences:**
Well... not really an ISSUE per se, more of strange space magic. 

**Steps to reproduce the problem**:
Right after round start, i eagerly click "Join Game" and choose botanist. Soon after that i began walking down hall, and before even making it to tool storage, i was whipped away by space magic all the way to botany, and got a second PDA with no id or cart..... so that's cool i guess....

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Extended game mode, Round number 533, server time round start, blahblahblah.

### LINKED ISSUE #753: Black screen + UI
example
**Problem Description**:
I joined the gamed the lobby screen loaded round was already going so I looked for a job found one joined but as soon as I joined I got a black screen but the UI was still there I cant right click to look at thing or move i think. But I can use UI elements move my pda turn on the light on it move the clip board and the like.

**What did you expect to happen**: 
Cause i couldnt move or see any game sprites.

**What happened instead**:
Basically its how i explained it.

**Why is this bad/What are the consequences:**
Keeps me from playing the round at all.

**Steps to reproduce the problem**:
I have no clue I joined in like normal the only thing I can think of thats diffrent was that i had AI as my highest job selection but that shouldnt matter.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: I was told to update byond and Direct X and I did update byond but Direct X is alredy in the latest version possible.

### LINKED ISSUE #754: Runtime in machinery.dm
image

### LINKED ISSUE #755: Runtime in priority_announce.dm, line 37: Cannot read null.toggles
image

### LINKED ISSUE #760: Invisible Monkeys persist in harassing some crew after leaving a habitable world.
What is the problem?
Got wrekd by monkeys I couldn't observe or interact with.

Why do you think this is a bug?
The new exoplanet code is pretty fresh, and there would of been more publicity about introducing hordes of Pred-Monkeys.

How is what happened different from what you expected?
I expected monkeys to at least reflect visible light.

Why do you think this is an important issue?
Kinda ruins the remainder of the round for whomever is affected. Cloning would probably reset monkey agro, but players shouldn't have to resort to that.

The most important section. Review everything you did leading up to causing the issue.
1. Ship lands on exoplanet with jungle world and multiude monkey hordes.
2. Engage monkey hordes in combat. 
3. Leave planet/system on the ship.
4. Await red walls of attack text

Anything else you can tell us.
image

I seemed to be the only one affected.
Monkeys didnt seem to care where I was or what z level I was on aside from when I took the q-pad back to the exoplanet.

### LINKED ISSUE #781: Lighting in lavalands tunnels is broken
**Problem Description**:
Lighting in lavalands tunnels is broken

**What did you expect to happen**: 
Lighting must be fine

**What happened instead**:
See pics

**Why is this bad/What are the consequences:**
Totally unplayable! Can't see anything at related turfs

**Steps to reproduce the problem**:
1. Fly to lavaland planet
2. Land at lavaland planet
3. Hope that it happens

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: Anything else you can tell us.
Round number: 554. Server time: ~14:40, today
ftl13 nsv astraeus 2017-05-06 213824
ftl13 nsv astraeus 2017-05-06 213925
ftl13 nsv astraeus 2017-05-06 213935

### LINKED ISSUE #804: telekenis body bag iseus
**Problem Description**:
If you unfold a body bag via telekenis it apears under you.

**What did you expect to happen**: 
I expected the body bag to unfold in maint.

**What happened instead**:
It apeared under me unfolded.

**Why is this bad/What are the consequences:**
For instance when you try to put someone in a body bag in space with telekenis this could be really annoying deadly even.

**Why is this bad/What are the consequences:**
Why do you think this is an important issue?

**Steps to reproduce the problem**:
1. Have telekenis.
2. Grab a bodybag with it.
3. Try to unfold it via telekenis.
4. Watch in confusion as it apears under you.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: I all had all the genetic superpowers. # #

### PR DESCRIPTION
**Pull request by ninjanomnom**
Closes issues related to tg code so we have a fresh issue tracker for rebase.

:cl: ninjanomnom, cirr, ike709, NIN, Vivalas
add: A complete rebase of ftl13 content into the most recent TG! Featuring exciting things like
add: Bugs
del: Bugs
tweak: Bugs
tweak: No more manifolds directly next to gas storages in atmosia.
tweak: Hydrogen is now supplied in canister form.
fix: Bugs
wip: Bugs
/:cl:

Issues(44):
closes #804 closes #781 closes #760 closes #755 closes #754 closes #753 closes #750 closes #749 closes #744 closes #742 closes #741 closes #722 closes #744 closes #707 closes #704 closes #703 closes #702 closes #701 closes #699 closes #674 closes #666 closes #651 closes #559 closes #554 closes #537 closes #503 closes #501 closes #488 closes #468 closes #453 closes #452 closes #445 closes #429 closes #416 closes #415 closes #393 closes #376 closes #343 closes #325 closes #312 closes #246 closes #243 closes #205 closes #112 closes #97

### CODE REVIEW

**Comment by CrAzYPiLoT-SS13** [comment]
Good luck handling the inevitable conflicts.

**Comment by optimumtact** [comment]
Yep it looks good to me

**Comment by IlyaIU** [comment]
Last time i checked there was 250 conflict files

**Comment by alexkar598** [comment]
Is tg not full on bugs?

**Comment by ninjanomnom** [comment]
It's 250 commits about, there are 884 conflicts

**Comment by ninjanomnom** [comment]
Alright the conflicts are now in the files themselves so we can deal with this one folder at a time

**Comment by IlyaIU** [comment]
hurray

**Comment by ninjanomnom** [comment]
397 conflicting files remaining, 1160 specific conflicts

**Comment by ninjanomnom** [comment]
https://puu.sh/vS7b7/45a724c7b9.png

**Comment by ike709** [comment]
It'll still probably be full of runtimes...

**Comment by ninjanomnom** [comment]
Not as bad as I feared, at least from the initialization runtimes.

File: code/modules/jobs/job_types/job.dm [hunk 1]
```diff
@@ -213,9 +213,9 @@
 
 	var/obj/item/weapon/card/id/C = H.wear_id
 
-	var/alt_title
-	if(H.mind)
-		alt_title = H.mind.role_alt_title
+	//var/alt_title
```
Comment (ike709) [inline code/modules/jobs/job_types/job.dm]: u wot m8

Comment (ninjanomnom) [inline code/modules/jobs/job_types/job.dm]: it's commented out for the moment because I'm dealing with other things first

**Comment by ninjanomnom** [comment]
It now passes travis, not much left to fix for it to be ready to merge.

File: code/modules/cargo/packs.dm [hunk 2]
```diff
@@ -2045,7 +2045,7 @@
 	cost = 800
 	contains = list(/obj/item/weapon/book/space_catalog)
 	crate_name = "space catalog crate"
-
+/*
 /datum/supply_pack/misc/civilian
 	name = "Civilian Cryocrew"
```
Comment (ike709) [inline code/modules/cargo/packs.dm]: rip cryocrew, you were hardly used

Comment (Qustinnus) [inline code/modules/cargo/packs.dm]: why is it being commented out? I still want to use them

**Comment by ike709** [comment]
WE DID IT! MISSION ACCOMPLISHED! WE FUCKING DID IT

**Comment by ninjanomnom** [comment]
godcanthear

**Pull request merged**